File: src/main/kotlin/StockData.kt
/**
 * 新浪行情接口返回的原始数据模型。
 * @param price 当前价格
 * @param preClose 昨日收盘价
 */
internal data class SinaQuote(val price: Double, val preClose: Double)

/**
 * 用于UI显示的股票数据模型。
 * @param price 当前价格
 * @param changePercent 涨跌幅
 * @param rise 瞬时涨跌幅 (与上次刷新相比)
 * @param indexPercent 大盘指数涨跌幅
 */
data class StockData(val price: Double, val changePercent: Double, val rise: Double, val indexPercent: Double)

File: rvan/src/data/model/StockData.ts
/**
 * 股票数据模型
 */
export interface StockData {
  code: string;
  name: string;
  price: number;
  changePercent: number;
  rise: number;
  indexPercent: number;
  timestamp: Date;
}

/**
 * 新浪行情原始数据
 */
export interface SinaQuote {
  price: number;
  preClose: number;
}

/**
 * 东方财富分时数据响应
 */
export interface EastMoneyTimeShareResponse {
  data?: TimeShareDataContainer;
}

/**
 * 分时数据容器
 */
export interface TimeShareDataContainer {
  code?: string;
  preClose: number;
  trends: string[];
}

/**
 * 分时图数据点
 */
export interface TimeSharePoint {
  time: string; // HH:mm 格式
  price: number;
}
